android/src/main/java/com/reactnativeone/OneModule.java

```diff
@@ -72,62 +84,84 @@ public void init(String siteKey, String touchpointUri, String apiKey,
         }
-      });
-    }
+        One.processResponse(response);
+        WritableNativeMap responseMap = responseObjectToReadableMap(response);
+        notifyResult(promise, responseMap);
+      } catch (ExecutionException error) {
+        notifyProblem(promise, Integer.toString(error.hashCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Completion Error: " + error.getCause());
+      } catch (OneSDKError error) {
+        notifyProblem(promise, Integer.toString(error.getSystemCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction SDK Error: " + error.getErrorMessage());
+      } catch (OneAPIError error) {
+        notifyProblem(promise, Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage());
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Api Error: " + error.getErrorMessage());
+      }
+    });
   }
 
   @ReactMethod
-  public void sendProperties(String interaction, ReadableMap propertiesMap) {
+  public void sendProperties(String interactionPath, ReadableMap propertiesMap) {
     HashMap<String, String> properties = getPropertiesFromReadableMap(propertiesMap);
-    final OneRequest sendPropertiesRequest = new OneRequest.Builder()
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
+    final OneRequest sendInteractionRequest = new OneRequest.Builder()
+      .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
       .properties(properties)
       .build();
-
-    final OneCall sendPropertiesCall = One.sendProperties(sendPropertiesRequest);
-    sendPropertiesCall.enqueue(null);
+    executor.submit(() -> {
+      try {
+        OneResponse response;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+          response = One.sendInteraction(true, sendInteractionRequest).join();
+        } else {
+          response = One.sendInteractionLegacySupport(true, sendInteractionRequest).join();
+        }
+        One.processResponse(response);
+      } catch (ExecutionException error) {
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Completion Error: " + error.getCause());
+      } catch (OneSDKError error) {
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction SDK Error: " + error.getErrorMessage());
+      } catch (OneAPIError error) {
+        Log.e(LOG_TAG, "[Thunderhead] Send Interaction Api Error: " + error.getErrorMessage());
+      }
+    });
   }
 
   @ReactMethod
-  public void sendResponseCode(String responseCode, String interaction) {
-    final OneResponseCodeRequest responseCodeRequest = new OneResponseCodeRequest.Builder()
-      .responseCode(new OneResponseCode(responseCode))
-      .interactionPath(new OneInteractionPath(URI.create(interaction)))
-      .build();
-    One.sendResponseCode(responseCodeRequest).enqueue(null);
+  public void sendResponseCode(String responseCode, String interactionPath) {
```

Comment:
Suggestion, be consistent and put interaction path first, just like the rest of the apis

Reply:
Good suggestion.  I did go off of the format in the iOS SDK on this one so if we want to update that one to be consistent as well let me know and I can write up a ticket.